Tests/StreamChatTests/Workers/ChannelUpdater_Tests.swift

```diff
@@ -429,6 +429,53 @@ final class ChannelUpdater_Tests: XCTestCase {
         XCTAssertEqual(channel?.messages.count, 4)
     }
 
+    func test_updateChannelQuery_whenSuccess_shouldCallOnBeforeSavingChannel() {
+        let query = ChannelQuery(cid: .unique)
+        let expUpdate = expectation(description: "update completes")
+        let expBeforeSaving = expectation(description: "should call before saving channel")
+
+        channelUpdater.update(
+            channelQuery: query,
+            isInRecoveryMode: false,
+            onBeforeSavingChannel: { _ in
+                expBeforeSaving.fulfill()
```

Comment:
Is it worth storing the timestamp when the call happened to make sure that beforeSaving is running before the save actually runs (basically test the separate db transaction that you have in your code)